libraries/postgres/README.md

```diff
@@ -314,9 +314,40 @@ An online migration may also be interrupted and replaced with an online downgrad
 
 ## Pagination
 
-Database functions which return many rows typically have `page_size int, page_offset int` as their last two arguments, and return a page of the given size at the given offset.
-The taskcluster-lib-api function `paginatedResults` is useful for translating such paginated results into an API response.
-For other users in Taskcluster services, this library provides `paginatedIterator` to convert paginated results into an async iterator.
+Pagination is the process of returning only some of the rows from a query, with a mechanism for getting the next "page" of rows in a subsequent query.
+The [taskcluster-lib-api function `paginateResults`](./api#pagination) is useful for translating such paginated results into an API response (and supports both types of pagination).
+
+### Index-Based
+
+The preferred mechanism for paginating database queries is to pass `page_size_in` giving the number of rows in a page, and one or more `after_.._in` parameters specifying where the page begins.
+The `after_.._in` parameters must correspond to an index on the table, so that Postgres can use that index to find the next row without a full scan.
+
+For other uses in Taskcluster services, this library provides `paginatedIterator` to convert paginated results into an async iterator.
+
+```javascript
+const {paginatedIterator} = require('taskcluster-lib-postgres');
+
+const doTheThings = async () => {
+  for await (let row of paginatedIterator({
+    indexColumns: ['foo_id', 'bar_id'],
+    fetch: async (page_size_in, after) => db.fns.get_widgets({
+      ...,
+      page_size_in,
+      ...after,
+    }),
+    size: 1000, // optional, defaults to 1000
+  })) {
+    // ..do something with `row`
+  }
+}
+```
+
+### Offset / Limit-Based
+
+Many database functions which return many rows have `page_size int, page_offset int` as their last two arguments, and return a page of the given size at the given offset.
```

Comment:
Should we mention the deprecation in the docs?

Reply:
I'm not sure.. I think the existing pagination is actually fine in a lot of cases, and as part of "stabilization" I don't want to kick off a long-term migration to index-based pagination.  That's why I wrote "preferred" :)